### PR TITLE
added flatten in tensorflow(New branch)

### PR DESCRIPTION
**Pull request by jgyfutub**
#19158

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    available_dtypes=helpers.get_dtypes(("valid"),
+    min_num_dims=3,
+        max_num_dims=5,
+        min_dim_size=1,
+        max_dim_size=4,
+        min_axis=-3,
+        max_axis=2),
+    test_with_out=st.just(False),
+    ))
+def test_tensorfloe_layers_flatten(
```
Comment (jshepherd01) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py:24]: Typo, should be `tensorflow`

**Review by jshepherd01 — CHANGES_REQUESTED** [review]
Looks like your tests have a few problems, you can run them to debug using `python3 -m pytest path/to/test/file.py::test_function_name -v`

File: ivy/functional/frontends/tensorflow/keras/layers.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import ivy
+import ivy.functional.frontends.tensorflow as tf_frontend
+from ivy.functional.frontends.tensorflow.func_wrapper import to_ivy_arrays_and_back
+
+@to_ivy_arrays_and_back
+def flatten(x,/,*,copy=None,start_dim=0, end_dim=-1, order='C', out=None):
```
[the remark this answers is not in the document]
Comment (jgyfutub) [reply ivy/functional/frontends/tensorflow/keras/layers.py:6]: I tooj help from link:
https://unify.ai/docs/ivy/docs/functional/ivy/experimental/manipulation/ivy.functional.ivy.experimental.manipulation.flatten.html

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    dtype_and_x=helpers.dtype_and_values(
+    available_dtypes=helpers.get_dtypes(("valid"),
+    min_num_dims=3,
+        max_num_dims=5,
+        min_dim_size=1,
+        max_dim_size=4,
+        min_axis=-3,
+        max_axis=2),
+    test_with_out=st.just(False),
+    ))
```
[the remark this answers is not in the document]
Comment (jgyfutub) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py:23]: no `min_num_dims`  are parameters to  ` helpers.dtype_and_values `

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py (hunk 1 above)
Comment (jgyfutub) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py:24]: yes i did a spelling mistake

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+@handle_frontend_test(
+    fn_tree="tensorflow.keras.layers.flatten",
+    dtype_and_x=helpers.dtype_and_values(
+    available_dtypes=helpers.get_dtypes(("valid"),
+    min_num_dims=3,
+        max_num_dims=5,
+        min_dim_size=1,
+        max_dim_size=4,
+        min_axis=-3,
+        max_axis=2),
```
[the remark this answers is not in the document]
Comment (jgyfutub) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py:21]: Actually I did a mistake there I just saw it in link:https://github.com/unifyai/ivy/blob/master/ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py , can you please suggest me a better way to tackle this?

**Comment by jgyfutub** [comment]
I have done some changes please review it @jshepherd01

File: ivy/functional/frontends/tensorflow/keras/layers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jshepherd01) [reply ivy/functional/frontends/tensorflow/keras/layers.py:7]: Ah, so the idea of the frontend functions is that they copy the structure of the frontend framework (in this case tensorflow) and internally call the ivy function. So rather than using the call signature for `ivy.flatten`, you should be using the call signature for `tensorflow.keras.flatten` and then calling the ivy function from within your function. Does that make sense?

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jshepherd01) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py]: Now that you've removed the spare opening bracket I think you're good on this one

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+@handle_frontend_test(
+    fn_tree="tensorflow.keras.layers.flatten",
+    dtype_and_x=helpers.dtype_and_values(
+    available_dtypes=helpers.get_dtypes("valid"),
+    min_num_dims=3,
+        max_num_dims=5,
+        min_dim_size=1,
+        max_dim_size=4,
+    ),
+     test_with_out=st.just(False))
```
Comment (jshepherd01) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py:21]: It looks like you've got some formatting errors here, you'll need to fix those by running the [pre-commit hooks](https://unify.ai/docs/ivy/overview/contributing/setting_up.html#pre-commit).

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    input_dtype, x  = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        backend_to_test=backend_fw,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        x=x[0],
+        )
```
Comment (jshepherd01) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py:40]: This doesn't seem to test the full range of what the function does, you should really be passing in a value for each of the function's arguments (`start_dim`, `end_dim`, and `order` at least)

**Review by jshepherd01 — CHANGES_REQUESTED** [review]

**Comment by jshepherd01** [comment]
Please also check [my response to the previous review](https://github.com/unifyai/ivy/pull/21502#discussion_r1293184015)

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py (hunk 6 above)
Comment (jgyfutub) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py:40]: but only dtype_and_x is giving only two variables input_dtype and x I tried using copy as extra parameter it was giving error

File: ivy/functional/frontends/tensorflow/keras/layers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jgyfutub) [reply ivy/functional/frontends/tensorflow/keras/layers.py:7]: yes I understood,thanks!!

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py (hunk 6 above)
Comment (jgyfutub) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_keras/test_layers.py:40]: okk thanks for helping!!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by jgyfutub** [comment]
I apologise since I haven't worked in the issue for a long time because I was busy in college exams and workload.Thus,I am still interested to work upon this issue.I will try to push some new commits in my pull request in 1-2 days.

**Comment by jgyfutub** [comment]
@DecFox I request you to please review my new commits

**Review by Sarvesh-Kesharwani — CHANGES_REQUESTED** [review]
Hi @thecoder12,
Let me know once you are done with the below suggested changes.

**Comment by Ishticode** [comment]
Closing this as requested changes were not made in two months and due to lack of activity. Feel free to reopen if you would like to continue working on this. Thank you very much for the efforts.
